src/features/swap/index.tsx

```diff
@@ -0,0 +1,207 @@
+  features: [SafeAppFeatures.BATCHED_TRANSACTIONS],
+  socialProfiles: [],
+}
+const SwapWidget = ({ sell }: Params) => {
+  const chainId = useChainId()
+  const { palette } = useTheme()
+  const darkMode = useDarkMode()
+  const dispatch = useAppDispatch()
+
+  const [toasts, setToasts] = useState<String[]>([])
+
+  const groupKey = 'swap-order-status'
+  const listeners = useMemo<CowEventListeners>(() => {
+    return [
+      {
+        event: CowEvents.ON_TOAST_MESSAGE,
+        handler: (event) => {
+          console.info('🍞 New toast message:', event)
+          const { message, messageType, data } = event
+          switch (messageType) {
+            // @ts-ignore
+            case 'ORDER_CREATED':
+              dispatch(
+                showNotification({
+                  title: 'Order created',
+                  message: 'Order created but waiting for presignature',
+                  groupKey,
+                  variant: 'info',
+                }),
+              )
+              break
+            // @ts-ignore
+            case 'ORDER_PRESIGNED':
+              dispatch(
+                showNotification({
+                  title: 'Order presigned waiting for fullfillment',
+                  message: 'Order was presigned and waiting for match',
+                  groupKey,
+                  variant: 'info',
+                }),
+              )
+              break
+            // @ts-ignore
+            case 'ORDER_FULFILLED':
+              dispatch(
+                showNotification({
+                  title: 'Order fullfilled',
+                  message: 'Order was fullfilled',
+                  groupKey,
+                  variant: 'info',
+                }),
+              )
+              break
+            // @ts-ignore
+            case 'ORDER_EXPIRED':
+              dispatch(
+                showNotification({
+                  title: 'Order expired',
+                  message,
+                  groupKey,
+                  variant: 'warning',
+                }),
+              )
+              break
+            // @ts-ignore
+            case 'ORDER_CANCELLED':
+              dispatch(
+                showNotification({
+                  title: 'Order cancelled',
+                  message,
+                  groupKey,
+                  variant: 'warning',
+                }),
+              )
+              break
+          }
+        },
+      },
+    ]
+  }, [dispatch])
+
+  const [params, setParams] = useState<CowSwapWidgetParams | null>(null)
+  useEffect(() => {
+    setParams({
+      appCode: 'Safe Wallet Swaps', // Name of your app (max 50 characters)
+      width: '100%', // Width in pixels (or 100% to use all available space)
+      height: '860px',
+      // provider: safeAppWeb3Provider, // Ethereum EIP-1193 provider. For a quick test, you can pass `window.ethereum`, but consider using something like https://web3modal.com
+      chainId, // 1 (Mainnet), 5 (Goerli), 100 (Gnosis)
```

Comment:
I think you don't need to provide this (if the provider is detected, it will ask the provider)

Reply:
doesn't seem to be the case. If I remove the chainId and let's say I'm on sepolia the page asks me to switch network to mainnet?

Reply:
My point is we should not hardcode this, as it will depend on the selected safe you use 

In the example, we didn't add it as param 
https://github.com/cowprotocol/safe-app-example/blob/main/src/components/App.tsx